### PR TITLE
Add support for GitHub registry cache

### PR DESCRIPTION
**Pull request by potiuk**
NOTE TO REVIEWERS! This one is based on top of #8496 so only review the last commit.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,14 +21,13 @@ language: python
...
 env:
   global:
-    - BUILD_ID=${TRAVIS_BUILD_ID}
     - MOUNT_LOCAL_SOURCES="false"
     - MOUNT_HOST_AIRFLOW_VOLUME="true"
     - FORCE_ANSWER_TO_QUESTIONS="yes"
     - SKIP_CHECK_REMOTE_IMAGE="true"
     - DB_RESET="true"
     - VERBOSE="true"
-    - CI="true"
```
Comment (potiuk) [inline .travis.yml:31]: NOTE ! CI is set to "true" by both Travis and Github. We do not need to set it separately.

File: scripts/ci/_utils.sh [hunk 2]
```diff
@@ -136,6 +136,14 @@ function initialize_common_environment {
             "--env" "PYTHONDONTWRITEBYTECODE" \
         )
     fi
+
+    if [[ ${CI:=} == "true" ]]; then
```
Comment (potiuk) [inline scripts/ci/_utils.sh:140]: For those curious ones. This is a recommended way of passing codecov variables to docker run command: https://docs.codecov.io/docs/testing-with-docker

**Review by kaxil — COMMENTED** [review]
Don't know much about Docker Caching. Can someone else review this PR please @dimberman  @turbaszek

File: scripts/ci/_utils.sh [hunk 3]
```diff
@@ -919,86 +933,82 @@ function match_files_regexp() {
...
+                export CI_TARGET_REPO="${GITHUB_REPOSITORY}"
+                export CI_TARGET_BRANCH="${GITHUB_BASE_REF}"
+            elif [[ ${GITHUB_EVENT_TYPE:=} == "schedule" ]]; then
+                export CI_EVENT_TYPE="schedule"
+                export CI_TARGET_REPO="${GITHUB_REPOSITORY}"
+                export CI_TARGET_BRANCH="${SOURCE_BRANCH}"
+            else
+                export CI_EVENT_TYPE="push"
+                export CI_TARGET_REPO="${GITHUB_REPOSITORY}"
+                export CI_TARGET_BRANCH="${SOURCE_BRANCH}"
```
Comment (turbaszek) [inline scripts/ci/_utils.sh]: Can we move those variables before elif? It seems we are replicating those lines.

File: scripts/ci/_utils.sh [hunk 4]
```diff
@@ -1071,31 +1127,15 @@ Docker pulling ${PYTHON_BASE_IMAGE}.
...
-            echo
-            if [[ -n ${DETECTED_TERMINAL:=""} ]]; then
-                echo -n "
-Docker pulling ${IMAGE}.
-" > "${DETECTED_TERMINAL}"
-            fi
-            verbose_docker pull "${AIRFLOW_CI_IMAGE}" | tee -a "${OUTPUT_LOG}" || true
-            echo
-        fi
+        # C production image
```
Comment (turbaszek) [inline scripts/ci/_utils.sh]:
```suggestion
        # production image
```
?

File: scripts/ci/_utils.sh (hunk 3 above)
Comment (potiuk) [reply scripts/ci/_utils.sh]: Or even after :) indeed.

File: scripts/ci/_utils.sh (hunk 4 above)
Comment (potiuk) [reply scripts/ci/_utils.sh]: Right :)

File: scripts/ci/_utils.sh [hunk 5]
```diff
@@ -919,86 +933,78 @@ function match_files_regexp() {
...
+            export CI_TARGET_REPO="${TRAVIS_REPO_SLUG}"
+            export CI_BUILD_ID="${TRAVIS_BUILD_ID}"
+            export CI_JOB_ID="${TRAVIS_JOB_ID}"
+        elif [[ ${GITHUB_ACTIONS:=} == "true" ]]; then
+            if [[ ${GITHUB_EVENT_NAME:=} == "pull_request" ]]; then
+                export CI_EVENT_TYPE="pull_request"
+            elif [[ ${GITHUB_EVENT_TYPE:=} == "schedule" ]]; then
+                export CI_EVENT_TYPE="schedule"
+            else
+                export CI_EVENT_TYPE="push"
```
Comment (turbaszek) [inline scripts/ci/_utils.sh:965]: I am not bash master but can't we simplify it?
```suggestion
            if [[ ${GITHUB_EVENT_NAME:=} != "" ]]; then
                export CI_EVENT_TYPE=${GITHUB_EVENT_NAME}
             else
                export CI_EVENT_TYPE="push"
```
or 
```bash
export CI_EVENT_TYPE=${GITHUB_EVENT_NAME:="push"}
```

**Review by turbaszek — APPROVED** [review]
Small comment but otherwise looks good

File: scripts/ci/_utils.sh (hunk 5 above)
Comment (potiuk) [reply scripts/ci/_utils.sh:965]: Yes. but I wanted to keep parallel with TRAVIS above -> this is just mapping from "CI specific" value to "common Ci value" so I prefer to keep the mapping.

**Pull request merged**